Add a debug menu

sajmon170 opened the pull request:
Currently adding new features is cumbersome because it's hard to test them. This pull request adds a simple debug menu that allows for
- faster map traversal by:
  - disabling collision
  - enabling debug movement
- faster ship prototyping by:
  - disabling money balance checks in the workshop
- easier map exploration by:
  - enabling invincibility mode
- Easy level testing by:
  - skipping directly to level select
  - disabling node validity checks.

Usage notes:
- The debug menu is only available on debug builds
- To use the debug menu press <F3> on your keyboard.
- The debug movement option remaps WASD keys to directly control the ship.

As a side effect, this pull request also fixes a bug where changing key bindings in the Input Map doesn't get reflected in key validity checks when binding keys to modules during ship building.

Lenetis reviewed (approved):
Everything is almost perfect - great job :)

The only major thing that I found is that when you turn on invincibility, all bosses also become invincible, as they use modules for their destructible parts. Maybe you could check if self.ship == GameManager.player_ship in the `take_damage` method?

It would also be nice to be able to press the buttons with your mouse, instead of having to use the keyboard.

A few minor bugs that are not 100% intuitive, but I don't thing they are worth spending much time on fixing:
- "Debug movement" without "Disable player collision" rotates the ship (it pulls the ship by the cockpit, instead of only translating it)
- "Disable player collisions" without "Debug movement" cause the ship to rotate very fast when turning
- "Enter map select screen" from the map select screen clears explored nodes
- "Disable money checks" also disables inventory checks (not a bad thing, but could be communicated somehow)

As for the code, the only nitpick I have is the `_ready()` method should always be placed before your own methods - see the suggested [code order](https://github.com/Cyberiada2024-2025/Faster-Than-Scrap?tab=readme-ov-file#code-order) (in `module.gd`)

sajmon170 commented:
Thanks for noticing the boss invincibility bug. I fixed it in the recent commit. I also fixed the code style issues you've mentioned (button description and function placement).

I was already aware of the weird behavior of disabling player collisions. However, one of the main goals of this branch was to keep the code changes to a bare minimum since this pull request affects many different GDScript modules at once. This includes not changing the logic of the release build.

I think that fixes to those issues should be done in separate branches since they might involve deeper changes of core release build logic.
